proposals/cleanup.md

```diff
@@ -233,6 +233,87 @@ Kyverno could alternatively leverage a CronJob resource to perform the deletions
 | The CronJobs can be easily removed from the cluster after the completion by setting their ownerReference to the resource it was created to delete. |  |
 
+## Finalized Implementation
+
+Finalized implementation is **Implementation 2 (reuse the Kubernetes CronJob resource)**. So, we will create new CRDs **CleanupPolicy** and **ClusterCleanupPolicy** for handling resource cleanup. 
+
+CleanupPolicy CRD manfiests example:
+```
+apiVersion: kyverno.io/v1alpha1
+kind: CleanupPolicy
+metadata:
+  name: cleanuppolicy
+  namespace: default
+spec:
+  match:
+    any:
+    - resources:
+        kinds:
+          - Pod
+  conditions:
+    any:
+    - key: "{{ request.name }}"
+      operator: Equals
+      value: example
+  schedule: "* * * * *"
+```
+A cleanup controller will reconcile these CRDs and create CronJobs which will execute commands to delete the specified resources. For implementing cleanup controller we have 2 options:
+
+
+### 1. Creating CronJobs per resource.
+A cleanup controller will be creating a CronJob for each of the matching resources. CronJob will run the kubectl command to delete these resources. CronJob will run according to the schedule specified in the cleanupPolicy. Creation and updation of a new cleanup policy will trigger the reconciliation and the reconciliation logic will get all the matching resources and will create a CronJob per resource which will have info and commands to delete the matching resource.
+
+
+
+**Flow:**
+
+1. A new policy is added to the cluster. 
+2. The cleanup policy object is added to the controller queue with the help of CleanupPolicy and ClusterCleanupPolicy informers.
+3. The controller then gets the policy.
+4. Get the matching resources for that rule/policy.
+5. Create CronJob for scheduling the deletion for each resource.
+6. Kyverno’s Service account is used in CronJob.
+7. OwnerReference of the CronJob is set to the trigger resource, so when the matching resource is deleted, the CJ is also deleted.
+
+**Life Cycle of the CronJob:**
+
+In this approach lifecycle of the CronJob created will be same as the matching resource. Since the OwnerReference of the matching resource will be set to the mathcing and this way CronJob's garbase collection will be handled.
+
+| Pros | Cons |
+| --- | --- |
+| Deletion of CJs after resource cleanup can be easily handled through OwnerReference | Creating CJs for each resource could be inefficient in large clusters. |
+
+
+### 2. Creating CronJobs per rule/policy.
+A cleanup controller will be creating CronJob per rule/policy. Creation and updation of a new cleanup policy will trigger the reconciliation. When reconciliation is triggered, a CronJob is created and delete commands for all the matching resources are informed to the CronJob and are executed at the same schedule (specified in the policy).
```

Comment:
..will ~~be creating~~ creates **a** CronJob .. Creation and ~~updation~~ **update**..